cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/ClusterBootstrap.scala

```diff
@@ -64,6 +64,26 @@ final class ClusterBootstrap(implicit system: ExtendedActorSystem) extends Exten
 
   private[this] val _selfContactPointUri: Promise[Uri] = Promise()
 
+
+  private val autostart = system.settings.config.getStringList("akka.extensions")
+    .indexOf(classOf[ClusterBootstrap].getName) > -1
+
+  if (autostart) {
+    log.info("ClusterBootstrap loaded through 'akka.extensions' auto starting management and bootstrap.")
+    // Akka Management hosts the HTTP routes used by bootstrap
+    Future {
+      // we can't let it block extension init
+      try {
+        AkkaManagement(system).start()
+        ClusterBootstrap(system).start()
+      } catch {
+        case ex =>
+          log.error("Failed to autostart cluster bootstrap, terminating system", ex)
+          system.terminate()
```

Comment:
Actually a failure would probably be async, bind port failure for example, so we'd need some way to catch that and terminate.

Reply:
AkkaManagement returns a Future so we could already call the ClusterBootstrap in a `map`. 

I will assume that a port binding error i AkkaManagement would result in a failed future. 

For the Bootstrap, if it is a remote port issue, it will crash immediately, no? That would fail the Future

Reply:
I think not mapping the two of them is fine, but we should capture the Future returned by `AkkaManagement#start()` and terminate system if it fails.`ClusterBootstrap#start()` sadly is `Unit` and starts some actor but not sure anything that should lead to termination can really go wrong there.